haystack/management/commands/update_index.py

```diff
@@ -84,10 +88,9 @@ def do_update(backend, index, qs, start, end, total, verbosity=1, commit=True,
                                                                              retries + 1,
                                                                              max_retries))
             break
+        except (SystemExit, KeyboardInterrupt):
+            raise
```

Comment:
Good point – I reworked that code several times while testing this and that's cruft